aether-kernel/aether/kernel/api/avro_tools.py

```diff
@@ -469,7 +468,7 @@ def avro_schema_to_passthrough_artefacts(item_id, avro_schema):
         # this is an auto-generated mapping that shouldn't be modified manually
         'is_read_only': True,
         'is_active': True,
-        'input': sample,  # include a data sample
+        'input': definition,  # include avro schema
```

Comment:
If this is the schema, for coherence we should rename it to schema, don't you think the same?

https://github.com/eHealthAfrica/aether/blob/develop/aether-ui/aether/ui/api/models.py#L32-L36

```python
class Pipeline(ExportModelOperationsMixin('ui_pipeline'), TimeStampedModel):
    # ...

    # this is the avro schema
    schema = JSONField(null=True, blank=True, default=dict)

    # this is an example of the data using the avro schema
    input = JSONField(null=True, blank=True, default=dict)
```

or have both :woman_shrugging:

Reply:
It's slightly annoying that you don't get randomly generated data with a schema that is shared with a read-only pipeline - would it be possible to have both?

Reply:
something like this?

```python
def avro_schema_to_passthrough_artefacts(item_id, avro_schema):
    # ...

    schema = {
        # ...
    }
    mappingset = {
        'id': item_id,
        'name': name,
        'schema': avro_schema,  # the original AVRO schema (without the missing id)
        'input': random_avro(avro_schema),
    }
    mapping = {
        'id': item_id,
        'name': name,
        'definition': {
            'entities': {name: item_id},
            'mapping': rules,
        },
        # this is an auto-generated mapping that shouldn't be modified manually
        'is_read_only': True,
        'is_active': True,
    }

    return schema, mappingset, mapping
```

Reply:
Yes, I guess so - it hadn't occurred to me that the generated data belongs to the `mappingset`, but that would actually be totally fine